storage/innobase/handler/ha_innodb.cc

```diff
@@ -19977,9 +19979,11 @@ ib_logf(
 		break;
 	case IB_LOG_LEVEL_ERROR:
 		sql_print_error("InnoDB: %s", str);
+		sd_notifyf(0, "STATUS=InnoDB: Error: %s", str);
 		break;
 	case IB_LOG_LEVEL_FATAL:
 		sql_print_error("InnoDB: %s", str);
+		sd_notifyf(0, "STATUS=InnoDB: Fatal: %s", str);
```

Comment:
Generally, sd_notify should only provide very rough levels of service status: starting up, running, shutting down, etc. Specific errors should go to the systemd journal (which syslog and stdour/stderr do by default); those errors will show as MariaDB's last words on systemctl status, anyway.